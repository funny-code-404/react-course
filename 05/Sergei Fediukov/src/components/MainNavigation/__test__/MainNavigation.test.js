import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import { MainNavigation } from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly "MainNavigation"', () => {
    const mockFn = jest.fn();

    const props = {
        getDataAction: mockFn,
        title: 'test title',
        path: 'test path'

    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<MainNavigation {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component have children div with "menu_link" class', () => {
        expect(wrapper.find('.menu_link')).toHaveLength(1)
    })
    it('component have children "NavLink"', () => {
        expect(wrapper.find('NavLink')).toHaveLength(1)
    })
    it('"NavLink" should call onClick when clicking', () => {
        wrapper.find('NavLink').simulate('click')
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
})