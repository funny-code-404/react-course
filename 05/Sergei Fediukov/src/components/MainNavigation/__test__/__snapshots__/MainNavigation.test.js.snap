// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "MainNavigation" to Match Snapshot 1`] = `
<div
  className="menu_link"
>
  <NavLink
    activeClassName="nav_active"
    exact={true}
    onClick={[Function]}
    to="test path"
  >
    test title
  </NavLink>
</div>
`;
